mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -53,24 +53,43 @@ import org.apache.spark.util.random.XORShiftRandom
  */
 private[recommendation] trait ALSModelParams extends Params with HasPredictionCol {
   /**
-   * Param for the column name for user ids.
+   * Param for the column name for user ids. Ids must be integers. Other
+   * numeric types are supported for this column, but will be cast to integers as long as they
+   * fall within the integer value range.
    * Default: "user"
    * @group param
    */
-  val userCol = new Param[String](this, "userCol", "column name for user ids")
+  val userCol = new Param[String](this, "userCol", "column name for user ids. Must be within " +
+    "the integer value range.")
 
   /** @group getParam */
   def getUserCol: String = $(userCol)
 
   /**
-   * Param for the column name for item ids.
+   * Param for the column name for item ids. Ids must be integers. Other
+   * numeric types are supported for this column, but will be cast to integers as long as they
```

Comment:
We "support" all numeric types in the sense that the input col can be any
numeric type. But it is cast to Int. It is a "safe" cast though, if it is >
Int.MaxValue or < Int.MinValue it throws an exception.

On Thu, 28 Apr 2016 at 21:08 Holden Karau <notifications@github.com> wrote:

> In mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala
> <https://github.com/apache/spark/pull/12762#discussion_r61485103>:
>
> >
> >    /** @group getParam */
> >    def getUserCol: String = $(userCol)
> >
> >    /**
> > -   * Param for the column name for item ids.
> > +   * Param for the column name for item ids. Ids must be integers. Other
> > +   * numeric types are supported for this column, but will be cast to integers as long as they
>
> So it seems like the only other numeric type we support is Long, maybe it
> would be better to say that? Someone might try and pass in BigInts or
> Doubles and expect this work.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/12762/files/73ea0b62f1c0ae6a9897ec83f5c8dfedea86f3f9#r61485103>
>